### ISSUE TITLE
Play All button

### ISSUE THREAD

**Issue by wildcard-f8** [FreeTubeApp/FreeTube#629]
Some channels I subscribe to upload multiple videos a day. When I play the newest video the next on the queue is usually from a totally different channel. So what I used to do in YouTube was go to their channel and click the Videos section and there's a Play All button. This would play their videos from the latest to the oldest so I don't have to go back after every video and look for the next one to play.

**Comment by GilgusMaximus** [comment]
This will come with the future playlist update, as we need an internal playlist for that. 
However it might take a while, because it is a major feature and PrestoN want to take his time with it

**Comment by JoAllg** [comment]
I just got FreeTube and this is a feature i am also missing. Will this really need an internal playlist? As far as i can see as a user on Youtube, the "Play all" link at a Channel is just a link to a playlist. And FreeTube is already able to play playlists.

At this point i would like to ask if you also have a NewPipe like "Background Playing"-Feature on the road map? This would be great when listening to music from playlists. I found it is  possible to choose "use audio formats" for a video, however this setting resets every video.

**Comment by shipaa** [comment]
As far i observe, the playall button is only for signed user. If that's correct the options is to use youtube api, but i don't know if that suitable for this case.

**Comment by GilgusMaximus** [comment]
No, we will basically grab the first page of videos and collect all id's and then throw them into a playlist and play that. No need for any API

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Possible solution

https://www.reddit.com/r/LifeProTips/comments/247c2u/lpt_youtube_how_to_play_all_videos_from_a_channel

Adding `?view=57` to the videos tab url brings back this feature

**Comment by ChunkyProgrammer** [comment]
"get the _channel id_ which starts
`UC`
then replace that with
`playlist?list=UU`
"
 @arrowgent

**Comment by UltraBlackLinux** [comment]
I intentionally picked the `?view=57` version, because some of the other stuff I've tried only shows ~100 videos at a time meaning that playlist shuffle only shuffles there. If another segment is loaded via just loading the next video, only this next segment is visible.
I don't know about the other solutions here, but it'd be good if you all checked if this is a thing with your solution

**Comment by seasidesand** [comment]
You can access the playlist, it seems, with this trick:
Get the channel ID, then change the first two characters from UC to UU, and attach it to https://www.youtube.com/playlist?list=

For example
.../channel/UCDO9bTgqwXx6cIq2_NMCvmA
becomes
https://www.youtube.com/playlist?list=UUDO9bTgqwXx6cIq2_NMCvmA
Which I can paste into the Freetube search bar! (Try it if you like.)

Ideally, I wouldn't have to do all that to open the channel's all upload playlist!
